src/System.Runtime/tests/System/GC.cs

```diff
@@ -226,4 +226,84 @@ private class TestObject
             }
         }
     }
-}
\ No newline at end of file
+
+    [Fact]
+    public static void ReRegisterForFinalize()
+    {
+        ReRegisterForFinalizeTest.Run();
+    }
+
+    private class ReRegisterForFinalizeTest
+    {
+        public static void Run()
+        {
+            TestObject.Finalized = false;
+            CreateObject();
+
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
```

Comment:
Adding @Maoni0 to look at these as I know in the past similar tests have been unreliable as depending on various aspects such as the JIT and such these will not be guaranteed be finialized.